src/lephare/process.py

```diff
@@ -219,3 +222,78 @@ def table_to_data(config, input, col_names=None, standard_names=False):
     flux[mask] = -99.0
     flux_err[mask] = -99.0
     return id, flux, flux_err, context, zspec, string_data
+
+
+def load_sed_list(path_to_list, object_type, absolute_paths=False):
+    """Take a sed list and move it with the sed files to the work directory.
+
+    We want to be able to load SED lists from other sources. These may use
+    absolute paths or relative to the list file. They must adopt the lepahre
+    standard which is relative to the sed/$TYPE/ directory
+
+    Parameters
+    ==========
+    path_to_list : str
+        The path to the SED list file.
+    object_type : str
+        The type of object. Must be one of "STAR", "GAL", or "QSO".
+    absolute_paths : bool
+        Set to True if list of absolute paths and not relative to list file
+
+    Returns
+    =======
+        new_list_file : str
+            The location of the new list file with respect to current
+            LEPHAREWORK directory. This is the value to set
+    """
+    # The type must by one of the three lephare standards
+    assert object_type in ["STAR", "GAL", "QSO"]
+    # The original file must exist
+    assert os.path.isfile(path_to_list)
+
+    # Set folder and file names.
+    name = path_to_list.split("/")[-1].split(".")[0]
+    old_dir = os.path.dirname(path_to_list)
+    new_dir = f"{lp.dm.lephare_dir}/sed/{object_type}/{name}"
+    new_list_file = f"{new_dir}/{name}.list"
+    # Create the config value that needs to be passed to lephare.
+    new_config_val = f"sed/{object_type}/{name}"
+    if os.path.exists(new_list_file):
+        print(
+            f"List file already exists at {new_list_file}\n"
+            "The SEDs may already have been loaded to the work directory."
+        )
+        return new_config_val
+
+    # Get the list of sed files.
+    with open(path_to_list, "r") as f:
+        lines = f.readlines()
+    sed_files = []
+    for line in lines:
+        # Ignore commented files
+        if line.strip()[0] != "#":
+            sed_files.append(line.replace("\n", ""))
+
+    # Make the folder to hold the seds.
+    assert not os.path.isdir(new_dir)
+    print(
+        f"Creating new sed directory at {new_dir}.\n" "Attempting to copy sed files and list files in place."
+    )
+    os.mkdir(new_dir)
```

Comment:
Maybe worth turning this into `os.mkdirs(new_dir)` so that the full directory structure can be created if needed.